### PR TITLE
adds void fluid chem

### PR DESCRIPTION
**Pull request by Adharainspace**
## About the PR 
adds a void fluid chem that turns floors and walls into simulated void walls and floors. it also has some effects when ingested you should find by testing it out! im not sure if this should be secret or not, so i left a placeholder recipe in. i tested it for errors and bugs and everything before submitting this.
https://forum.ss13.co/showthread.php?tid=13628 (old patch thread)

## Why's this needed? 
more chems are always more fun, and i think this adds something gimmicky and unique that doesnt really exist in game yet. 

## Changelog

```
adharainspace:
* adds void fluid chem! try drinking it and putting it on floors!
```

### CODE REVIEW

File: code/modules/chemistry/Reagents-Misc.dm [hunk 1]
```diff
@@ -3917,6 +3917,58 @@ datum
...
+				else
+					return
+
+			on_mob_life(var/mob/M, var/mult = 1)
+				if (!M) M = holder.my_atom
+				if ((holder.get_reagent_amount(src.id) >= 10) && prob(15)) //small(ish) prob to puke on life
+					var/turf/T = get_turf(M)
+					if (!locate(/obj/decal/cleanable/voidpuke) in T)
+						playsound(T, "sound/impact_sounds/Slimy_Splat_1.ogg", 50, 1)
+						new /obj/decal/cleanable/voidpuke(T)
```
Comment (mybluecorners) [inline code/modules/chemistry/Reagents-Misc.dm:3959]: use the make_cleanable proc instead
```suggestion
						make_cleanable(/obj/decal/cleanable/voidpuke,T)
```

**Comment by pali6** [comment]
I'd personally like to hear more opinions about the effect. I think it's cool and neat but at the same time I feel like it might be a bit too disruptive / annoying visually.

**Comment by ZeWaka** [comment]
This seems quite abusable.

File: code/modules/chemistry/Reagents-Misc.dm (hunk 1 above)
Comment (Adharainspace) [reply code/modules/chemistry/Reagents-Misc.dm]: i made this change and tested it out, it works

**Comment by Adharainspace** [comment]
> This seems quite abusable.
do you have any suggestions? what specifically could be abuseable?

**Comment by pali6** [comment]
Oh, it actually changes the turf instead of just changing the appearance, right? Hmm. 
- you prevent walls from being deconstructable by smoking / splashing this
- you remove the ability to pry off the cover of floor tiles
I'd personally make the turf effects visual only. Adding an overlay would pobably be fine. You could even play around with `blend_mode` and `alpha` to create an effect that mixes the original look of the turf with the void turf.

**Comment by Adharainspace** [comment]
im gonna close this, i dont care enough to put the time in i would need to to make this better

**Pull request closed without merging**